src/esi/Expression.cc

```diff
@@ -62,16 +62,23 @@ typedef enum {
     ESI_LITERAL_INVALID
 } literalhint;
 
-struct _stackmember {
-    evaluate *eval;
+struct stackmember {
+    evaluate *eval = nullptr;
     union {
         char *string;
         double floating;
         int integral;
     } value;
-    literalhint valuestored;
-    evaltype valuetype;
-    int precedence;
+    literalhint valuestored = ESI_LITERAL_INVALID;
+    evaltype valuetype = ESI_EXPR_INVALID;
+    int precedence = 0;
+
+    stackmember() :
+        eval(nullptr), valuestored(ESI_LITERAL_INVALID),
+        valuetype(ESI_EXPR_INVALID), precedence(0)
+    {
+        memset(&value, 0, sizeof(value));
+    }
 };
```

Comment:
> PR description: Add default initializer for POD struct stackmember in ESI, to ensure all its fields are initialised in all instantiations

The current PR description does not quite match the current code.

Reply:
Changed. Note that at least one c++ linter complains about memsetting a union that contains a double. I don't think we need to be concerned by this

Feel free to change the PR description if you feel it is not adequate

Reply:
> Feel free to change the PR description if you feel it is not adequate

Thank you for adjusting it. The current version matches the code so I left it as is after adding a Coverity defect type ID -- `UNINIT` -- something stable that is easier to search for than the textual description of that type.